synapse/handlers/register.py

```diff
@@ -195,16 +200,12 @@ async def check_username(
                         errcode=Codes.FORBIDDEN,
                     )
 
-        if guest_access_token is None:
-            try:
-                int(localpart)
```

Comment:
hah, can't believe this is how we defined the grammar for guest localparts. Goodness help us if Python allowed things like `0x42` in `int()`!

Reply:
I did try `int("+3")` in a Python shell after seeing that :angel:

Reply:
> hah, can't believe this is how we defined the grammar for guest localparts. Goodness help us if Python allowed things like `0x42` in `int()`!

It did not, but it did allow things like `042`. The new behavior should match that.

It did also allow `+42` and `-42` those will no longer be allowed.